controllers/helmrelease_controller.go

```diff
@@ -402,8 +403,13 @@ func (r *HelmReleaseReconciler) reconcileRelease(ctx context.Context, log logr.L
 
 			// Propagate any test error if not marked ignored.
 			if testErr != nil && !remediation.MustIgnoreTestFailures(hr.Spec.GetTest().IgnoreFailures) {
-				testsPassing := meta.GetCondition(hr.Status.Conditions, v2.TestSuccessCondition)
-				v2.SetHelmReleaseCondition(&hr, v2.ReleasedCondition, corev1.ConditionFalse, testsPassing.Reason, testsPassing.Message)
+				testsPassing := apimeta.FindStatusCondition(hr.Status.Conditions, v2.TestSuccessCondition)
+				apimeta.SetStatusCondition(&hr.Status.Conditions, metav1.Condition{
+					Type:    v2.ReleasedCondition,
+					Status:  metav1.ConditionFalse,
+					Reason:  testsPassing.Reason,
+					Message: testsPassing.Message,
+				})
```

Comment:
Don't we want to use https://github.com/fluxcd/pkg/pull/49 here so that we will get the observed generation handling once it is implemented?